adam_core/coordinates/origin.py

```diff
@@ -0,0 +1,106 @@
+import logging
+from enum import Enum
+from typing import Optional
+
+import numpy as np
+import pyarrow as pa
+import pyarrow.compute as pc
+from quivr import StringField, Table
+
+from ..constants import KM_P_AU, S_P_DAY
+
+logger = logging.getLogger(__name__)
+
+
+class OriginCodes(Enum):
+    SOLAR_SYSTEM_BARYCENTER = 0
+    MERCURY_BARYCENTER = 1
+    VENUS_BARYCENTER = 2
+    EARTH_MOON_BARYCENTER = 3
+    MARS_BARYCENTER = 4
+    JUPITER_BARYCENTER = 5
+    SATURN_BARYCENTER = 6
+    URANUS_BARYCENTER = 7
+    NEPTUNE_BARYCENTER = 8
+    SUN = 10
+    MERCURY = 199
+    VENUS = 299
+    EARTH = 399
+    MOON = 301
+    MARS = 499
+    JUPITER = 599
+    SATURN = 699
+    URANUS = 799
+    NEPTUNE = 899
+
+
+def _convert_mu_units(mu: float):
+    """
+    Convert mu in km^3 / s^2 to au^3 / day^2.
+    """
+    return mu / KM_P_AU**3 * S_P_DAY**2
+
+
+class OriginGravitationalParameters(float, Enum):
+    # Taken from https://iopscience.iop.org/article/10.3847/1538-3881/abd414
+    MERCURY_BARYCENTER = _convert_mu_units(22032.080486418)
+    VENUS_BARYCENTER = _convert_mu_units(324858.592000)
+    MARS_BARYCENTER = _convert_mu_units(42828.375816)
+    JUPITER_BARYCENTER = _convert_mu_units(126712764.100000)
+    SATURN_BARYCENTER = _convert_mu_units(37940584.841800)
+    URANUS_BARYCENTER = _convert_mu_units(5794556.400000)
+    NEPTUNE_BARYCENTER = _convert_mu_units(6836527.100580)
+    SUN = _convert_mu_units(132712440041.279419)
+    MERCURY = _convert_mu_units(22031.868551)
+    VENUS = _convert_mu_units(324858.592000)
+    EARTH = _convert_mu_units(398600.435507)
+    MOON = _convert_mu_units(4902.800118)
+
+
+# TODO: Replace with DictionaryField or similar
+#       Investigate whether this class is even necessary
+class Origin(Table):
+
+    code = StringField(nullable=False)
```

Comment:
Similar to `Frames`, we do end up actually building in some functionality for gravitational parameter. I think I'll keep it as is for now. This will need some thought for observatories. Hoping to start that work tomorrow.